apps/api/src/app/auth/services/auth.service.ts

```diff
@@ -89,20 +93,45 @@ export class AuthService {
         const withoutUsername = user.tokens.find(
           (i) => i.provider === AuthProviderEnum.GITHUB && !i.username && String(i.providerId) === String(profile.id)
         );
-
-        if (withoutUsername) {
-          await this.userRepository.update(
-            {
-              _id: user._id,
-              'tokens.providerId': profile.id,
-            },
-            {
-              $set: {
-                'tokens.$.username': profile.login,
+        // Checks if user's github primary email already exists but from a different authentication provider.
+        const withoutToken = user.tokens.findIndex((i) => i.provider === AuthProviderEnum.GITHUB) === -1;
+        const userUpdates: Promise<any>[] = [];
+
+        if (withoutUsername || email !== user.email) {
+          userUpdates.push(
+            this.userRepository.update(
+              {
+                _id: user._id,
+                'tokens.providerId': profile.id,
```

Comment:
Following the next if statement, could there be a case when the user is withoutUsername AND withoutToken? if it could be I think this query will return an empty result.

Reply:
Given we already have this condition
https://github.com/novuhq/novu/pull/3443/files#diff-7b90561f1765cc8061f3065abde564abfdf3891dac81d2e99577cd44285b8061R92

It will update the user info if any of the below holds true
https://github.com/novuhq/novu/pull/3443/files#diff-7b90561f1765cc8061f3065abde564abfdf3891dac81d2e99577cd44285b8061R100

Reply:
@djabarovgeorge 

Given we already have this condition
https://github.com/novuhq/novu/blob/5e1a23fdaa143d5bc23ace875d34f35afd8ce3b2/apps/api/src/app/auth/services/auth.service.ts#L92

It will update the user info if any of the below condition is the case
https://github.com/novuhq/novu/blob/5e1a23fdaa143d5bc23ace875d34f35afd8ce3b2/apps/api/src/app/auth/services/auth.service.ts#L100